wrapper/src/main/java/com/moodysalem/phantomjs/wrapper/PhantomJS.java

```diff
@@ -121,16 +137,28 @@ public static InputStream render(final PhantomJSOptions options, final InputStre
         Files.copy(new ByteArrayInputStream(footerInfo.getGeneratorFunction().getBytes()), footerFunctionPath);
 
         // the output file
-        final Path renderPath = PhantomJSConstants.TEMP_RENDER_DIR.resolve(String.format(PhantomJSConstants.TARGET_PREFIX + "%s.%s", renderId, renderFormat.name().toLowerCase()));
-
-        final PhantomJSExecutionResponse phantomJSExecutionResponse = exec(renderScript, options, new CommandLineArgument(paperSize.getWidth()), new CommandLineArgument(paperSize.getHeight()), new CommandLineArgument(dimensions.getWidth()),
-                new CommandLineArgument(dimensions.getHeight()), new CommandLineArgument(margin.getTop()), new CommandLineArgument(margin.getRight()), new CommandLineArgument(margin.getBottom()), new CommandLineArgument(margin.getLeft()),
-                new CommandLineArgument(headerInfo.getHeight()), new CommandLineArgument(wrapCommandLineArgumentName(PhantomJSConstants.HEADERFUNCTION_FILE), PhantomJSConstants.HEADERFUNCTION_FILE, headerFunctionPath.toFile()),
-                new CommandLineArgument(footerInfo.getHeight()), new CommandLineArgument(wrapCommandLineArgumentName(PhantomJSConstants.FOOTERFUNCTION_FILE), PhantomJSConstants.FOOTERFUNCTION_FILE, footerFunctionPath.toFile()),
-                new CommandLineArgument(OperatingSystem.get().name()), new CommandLineArgument(wrapCommandLineArgumentName(PhantomJSConstants.SOURCEPATH_TEMPLATENAME), PhantomJSConstants.SOURCEPATH_TEMPLATENAME, sourcePath.toFile()),
-                new CommandLineArgument(wrapCommandLineArgumentName(PhantomJSConstants.RENDERPATH_TEMPLATENAME), PhantomJSConstants.RENDERPATH_TEMPLATENAME, renderPath.toFile()),
-                new CommandLineArgument(wrapCommandLineArgumentName(PhantomJSConstants.JSWAIT_TEMPLATENAME), PhantomJSConstants.JSWAIT_TEMPLATENAME, jsWait),
-                new CommandLineArgument(wrapCommandLineArgumentName(PhantomJSConstants.JSINTERVAL_TEMPLATENAME), PhantomJSConstants.JSINTERVAL_TEMPLATENAME, jsInterval));
+        final Path renderPath = PhantomJSConstants.TEMP_RENDER_DIR.resolve(String.format(PhantomJSConstants.TARGET_PREFIX + "%s.%s", renderId, options.getRenderFormat().name().toLowerCase()));
+
+        List<CommandLineArgument> commandLineArguments = new ArrayList<>();
```

Comment:
I agree. I could change that to a fixed array instead of the List and conversion.
Regarding the final: I totally agree. I switched to intellij and I didn't activate the saveActions plugin. This comes for free in eclipse. ;)